src/interface.jl

```diff
@@ -83,6 +95,23 @@ function promote_operation_fallback(
     return promote_operation(new_op, T, z)
 end
 
+function promote_operation_fallback(
+    op::AddSubMul,
+    ::Type{S},
+    ::Type{LinearAlgebra.UniformScaling{T}},
+) where {S,T}
+    return promote_operation(op, S, T)
```

Comment:
How is this related to this PR ?

Reply:
`promote_type(Number, LinearAlgebra.UniformScaling{Bool})` ends up as `Any`, which is incorrect.

Reply:
Actually, this is avoidable